server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -55,9 +61,31 @@
 @Produces(Constants.JSON_CONTENT_TYPE)
 public class DeletedMessagesVaultRoutes implements Routes {
 
+    enum Action {
+        RESTORE;
+
+        static Action getAction(String actionString) {
+            Preconditions.checkArgument(StringUtils.isNotBlank(actionString), "action cannot be null or empty");
+
+            return Stream.of(values())
+                .filter(action -> action.name().equalsIgnoreCase(actionString))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(
+                    "'" + actionString + "' is not a valid action(" + String.join(",", actionStrings()) + ")"));
+        }
+
+        private static List<String> actionStrings() {
```

Comment:
I then prefer passing the string as a parameter of the enum rather then relying on the name

```
RESTORE("restore");

private final String name;

// constructor + getter
```

The code will become less sensible to refactoring and we wil be able to handle non-lowercase inputs.

BTW you did not provide a local for the toLowerCase